jdrfandroidbleparser/src/test/java/org/ehealthinnovation/jdrfandroidbleparser/bgm/characteristic/compoundcharacteristic/GlucoseFeatureCharacteristicTest.kt

```diff
@@ -0,0 +1,75 @@
+package org.ehealthinnovation.jdrfandroidbleparser.bgm.characteristic.compoundcharacteristic
+
+import org.ehealthinnovation.jdrfandroidbleparser.BaseTest
+import org.ehealthinnovation.jdrfandroidbleparser.encodedvalue.GattCharacteristic
+import org.ehealthinnovation.jdrfandroidbleparser.encodedvalue.bgm.feature.Flags
+import org.junit.Assert
+import org.junit.Assert.*
+import org.junit.Test
+import java.util.*
+
+class GlucoseFeatureCharacteristicTest : BaseTest() {
+
+    @Test
+    fun testTag() {
+        val glucoseFeatureCharacteristic = GlucoseFeatureCharacteristic(null)
+        Assert.assertEquals(GlucoseFeatureCharacteristic::class.java.canonicalName as String, glucoseFeatureCharacteristic.tag)
+    }
+
+
+    @Test
+    fun testAssignedNumber() {
+        val glucoseFeatureCharacteristic = GlucoseFeatureCharacteristic(null)
+        Assert.assertEquals(GattCharacteristic.GLUCOSE_FEATURE.assigned, glucoseFeatureCharacteristic.uuid)
+    }
+
+    @Test
+    fun testPartialParseFail() {
+        // This is a garbage payload, with no readable data, in terms of successfully parsing Glucose Context
+        val badPayload = byteArrayOf(Byte.MIN_VALUE)
+        val glucoseFeatureCharacteristic = GlucoseFeatureCharacteristic(mockBTCharacteristic(badPayload))
+
+        Assert.assertFalse(glucoseFeatureCharacteristic.successfulParsing)
+    }
+
+    @Test
+    fun parseSuccess() {
+        /**This is a general normal packet*/
+        val testPacket: ByteArray = byteArrayOf(0b00010001, 0b00000001)
```

Comment:
Why change the way we construct the byte array? Structure it the same as in the other tests

Reply:
Since it is a flag field, constructing it with 0b format can clearly show which bit is set.